server/static/upload.js

```diff
@@ -1,137 +1,242 @@
-        progress.remove();
-        return false;
+    // Add files to a specific sample
+    $('.table-data').on('click', '.add-files', function(e) {
+        sampleName = $(this).parents('tr').find('.sample-name').text();
     });
-    $('body').on('click', '.progress-retry-link', function() {
-        // Retry upload of this file
-        var progress = $(this).closest('.resumable-progress');
-        var uniqueIdentifier = progress.attr('data-file-id');
-        r.getFromUniqueIdentifier(uniqueIdentifier).retry();
 
-        progress.find('.progress-bar').first()
-            .removeClass('progress-bar-warning progress-bar-danger')
-            .addClass('progress-bar-striped active')
-            .html('Retrying...');
-        progress.find('.progress-cancel-link').show();
-        progress.find('.progress-retry-link').hide();
-        return false;
+    // Edit a sample name in table
+    $('.table-data').on('click', '.edit-table-sample-name', function(e){
+        sampleNode = $(this).parents('tr').find('.sample-name');
+        currentName = sampleNode.text()
+        sampleNode.replaceWith(function(){
+            return $('<td class="sample-name"><form class=edit-sample-name-form>'+
+            '<input id="sample-table-text" type="text" name="sampleName" class="form-control"'+
+            'placeholder="'+currentName+'"></form></td>')
+            });
     });
-    $('.progress-resume-link').on('click', function() {
-        // Resume upload of all files
-        r.upload();
-        // Show 'pause' button, hide 'resume' button
-        $('.progress-pause-link').show();
-        $('.progress-resume-link').hide();
-        return false;
+    $('.table-data').on('submit', '.edit-sample-name-form', function(e){
+            newName = $(this).find('#sample-table-text').val();
+            tbodyElem = $(this).parents('tbody');
+            $(this).remove();
+            tbodyElem.attr('name', newName).find('.sample-name').html(newName);
+            return false;
     });
-    $('.progress-pause-link').on('click', function() {
-        // Pause upload of all files
-        r.pause();
-        // Show 'resume' button, hide 'pause' button
-        $('.progress-pause-link').hide();
-        $('.progress-resume-link').show();
-        return false;
+
+    // Collapse the table contents and show only the panel header
+    $('.panel-heading').on('click', function(e) {
+         if ($(this).find('.table-collapse').hasClass('glyphicon-triangle-right') == true) {
+             $(this).parents('.panel').find('.panel-body').hide();
+             $(this).parents('.panel').find('.table').hide();
+             $(this).find('.table-collapse').removeClass('glyphicon-triangle-right').addClass('glyphicon-triangle-bottom');
+         } else if ($(this).find('.table-collapse').hasClass('glyphicon-triangle-bottom') == true) {
+             $(this).parents('.panel').find('.panel-body').show();
+             $(this).parents('.panel').find('.table').show();
```

Comment:
1. Would suggest `closest`
2. Would suggest refactoring out some of this redundancy. E.g. putting this outside of the if statement: `var $panel = $(this).parents('.panel')`